### PR TITLE
Feature: Add support for emitting SBOMs

### PR DESCRIPTION
**Pull request by mattmoor**
:gift: These are surfaced via a new output `sboms` which is a map from the image digest to it's SBOM.

depends on https://github.com/chainguard-dev/terraform-provider-apko/pull/42

/kind feature

### CODE REVIEW

File: internal/provider/resource_build_test.go [hunk 1]
```diff
@@ -48,12 +49,32 @@ archs:
...
+						ms := s.RootModule()
+						rs, ok := ms.Outputs["bar"]
+						if !ok {
+							return fmt.Errorf("Not found: %s", "bar")
+						}
+						sboms, ok := rs.Value.(map[string]interface{})
+						if !ok {
+							return fmt.Errorf("Incorrect type %T", rs.Value)
+						}
+						if len(sboms) != 3 {
```
Comment (mattmoor) [inline internal/provider/resource_build_test.go:73]: One for each architecture and the index.

File: internal/provider/build.go [hunk 2]
```diff
@@ -30,56 +31,56 @@ func fromImageData(data BuildResourceModel, wd string) (*build.Context, error) {
 	}
 	opts = append(opts,
 		build.WithImageConfiguration(ic),
-		// TODO(mattmoor): SBOMs would be nice
+		build.WithSBOMFormats([]string{"spdx"}),
```
Comment (mattmoor) [inline internal/provider/build.go:34]: We _**could**_ make this configurable, but I'm not really sure where our CycloneDX support stands, and so I am biasing towards leaving this.  What do folks think?

**Comment by mattmoor** [comment]
I am able to have this produce SBOM attestations with the following:
```
resource "cosign_attest" "foo" {
  for_each = apko_build.foo.sboms

  image          = each.key
  predicate_type = "https://spdx.dev/Document"
  predicate      = each.value
}
```

image

File: internal/provider/build.go (hunk 2 above)
Comment (jdolitsky) [reply internal/provider/build.go:34]: leave

**Review by jdolitsky — APPROVED** [review]

**Pull request merged**